pkg/localcontroller/manager/dataset.go

```diff
@@ -140,28 +153,37 @@ func (dm *DatasetManager) monitorDataSources(name string) {
 		return
 	}
 
-	if ds.Spec.URL == "" {
-		klog.Errorf("dataset(name=%s) empty data source url.", name)
+	dataURL := ds.Spec.URL
+	prefix, err := storage.CheckURL(dataURL)
+	if err != nil {
+		klog.Errorf("dataset(name=%) has the invalid url, error: %+v", name, err)
 		return
 	}
 
+	if prefix == storage.LocalPrefix {
```

Comment:
as said above, better to place these storage logic into storage.go

Reply:
fix it